GithubHandler.py

```diff
@@ -139,16 +155,36 @@ def get_pr_deltas(self, pr):
         Returns:
             str: The deltas for the pull request.
         """
+
+        valid_extensions = self.FILE_EXTENSIONS
         files_and_deltas = ""
         file_changes = pr.get_files()
         for f in file_changes:
+            if not any([f.filename.endswith(ext) for ext in valid_extensions]):
+                continue
             file_full_content = self.get_file_contents(f.filename, ref=pr.head.sha)
             full_file_delta = self.get_full_file_delta(f.filename, file_full_content)
-            file_delta = self.get_file_delta(f.patch, f.filename)
-            files_and_deltas += full_file_delta + "\n" + file_delta + "\n"
+            file_delta_str = self.get_file_delta(f.patch, f.filename)
+            files_and_deltas += full_file_delta + "\n" + file_delta_str + "\n"
```

Comment:
Duplication of `self.FILE_EXTENSIONS` could be avoided by referencing the class attribute directly rather than reassigning it to a local variable.
```suggestion
        if not any(f.filename.endswith(ext) for ext in GithubHandler.FILE_EXTENSIONS):
```
        As a further optimization, consider creating a set from `FILE_EXTENSIONS` for O(1) lookups.